src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
It's too looooong a callback
So, let's divide this
We should create a new function that checks if the child has a mother
after that, we can decrease our code & make this code more readable

```suggestion
 const filteredByChildren = people.filter(person => onlyWithSon
    ? person.sex === 'm' & checkIsHasMom(person.mother, people)
    : checkIsHasMom(person.mother, people)
  );
```